video/gazebocamerainput.h

```diff
@@ -0,0 +1,134 @@
+#pragma once
+
+// BoB robotics includes
+#include "common/assert.h"
+#include "input.h"
+
+// Gazebo includes
+#include <gazebo/gazebo_config.h>
+#include <gazebo/transport/transport.hh>
+#include <gazebo/msgs/msgs.hh>
+#include <gazebo/common/common.hh>
+
+// Gazebo's API has changed between major releases. These changes are
+// accounted for with #if..#endif blocks in this file.
+#if GAZEBO_MAJOR_VERSION < 6
+#include <gazebo/gazebo.hh>
+#else
+#include <gazebo/gazebo_client.hh>
+#endif
+
+// OpenCV
+#include <opencv2/opencv.hpp>
+
+// Standard C++ includes
+#include <stdexcept>
+#include <mutex>
+#include <atomic>
+
+namespace BoBRobotics {
+namespace Video {
+
+constexpr const char *GazeboCameraDeviceName = "gazebo_camera";
+
+//----------------------------------------------------------------------------
+// BoBRobotics::Video::GazeboCameraInput
+//----------------------------------------------------------------------------
+//! A thin wrapper for reading from any Gazebo camera feed
+class GazeboCameraInput : public Input
+{
+public:
+    /*!
+     * \brief Create a video stream using a new transport node and default topic
+     */
+    GazeboCameraInput()
+    : GazeboCameraInput("/gazebo/default/camera/link/camera/image")
+    {}
+    /*!
+     * \brief Create a video stream using a new transport node
+     */
+    GazeboCameraInput(const std::string &topic, const std::string &cameraName = GazeboCameraDeviceName)
+    : m_CameraTopic(topic), m_CameraName(cameraName)
+    {
+        // Load gazebo as a client
+        #if GAZEBO_MAJOR_VERSION < 6
+        gazebo::setupClient(0, 0);
+        #else
+        gazebo::client::setup(0, 0);
+        #endif
+        //Setup image subscription node
+        m_ImageNode= gazebo::transport::NodePtr(new gazebo::transport::Node());
+        m_ImageNode->Init();
+        subscribeToGazeboCamera();
+    }
+    /*!
+     * \brief Create a video stream for a Gazebo topic using a user-provided tranport node
+     *
+     * @param node Gazebo transport node
+     * @param topic Gazebo transport topic on which to subscribe
+     */
+    
+    GazeboCameraInput(gazebo::transport::NodePtr node, const std::string &topic="/gazebo/default/camera/link/camera/image", const std::string &cameraName = GazeboCameraDeviceName)
+      : m_CameraTopic(topic), m_CameraName(cameraName), m_ImageNode(node)
+    {
+        subscribeToGazeboCamera();
+    }
+
+    void subscribeToGazeboCamera(){
+        // Subscribe to the topic, and register a callback
+        m_HaveReceivedFrames = ATOMIC_VAR_INIT(false);
```

Comment:
I think one could just write ``m_HaveReceivedFrames = false;``. ``ATOMIC_VAR_INIT`` is just for backwards compatibility with C.